src/configure/configure.ts

```diff
@@ -177,73 +181,131 @@ class PipelineConfigurer {
     private async getSourceRepositoryDetails(): Promise<void> {
         try {
             if (!this.workspacePath) { // This is to handle when we have already identified the repository details.
-                let sourceOptions: Array<QuickPickItem> = [];
-                if (vscode.workspace && vscode.workspace.rootPath) {
-                    sourceOptions.push({ label: SourceOptions.CurrentWorkspace });
-                }
-
-                sourceOptions.push({ label: SourceOptions.BrowseLocalMachine });
-
-                let selectedSourceOption = await this.controlProvider.showQuickPick(
-                    sourceOptions,
-                    { placeHolder: Messages.selectFolderOrRepository }
-                );
-
-                this.telemetryHelper.setTelemetry(TelemetryKeys.SourceRepoLocation, selectedSourceOption.label);
-                switch (selectedSourceOption.label) {
-                    case SourceOptions.BrowseLocalMachine:
-                        let selectedFolder: vscode.Uri[] = await vscode.window.showOpenDialog(
-                            {
-                                openLabel: Messages.selectLabel,
-                                canSelectFiles: false,
-                                canSelectFolders: true,
-                                canSelectMany: false
-                            }
-                        );
-                        if (selectedFolder && selectedFolder.length > 0) {
-                            this.workspacePath = selectedFolder[0].fsPath;
-                        }
-                        else {
-                            throw new Error(Messages.noWorkSpaceSelectedError);
-                        }
-                        break;
-                    case SourceOptions.CurrentWorkspace:
-                        this.workspacePath = vscode.workspace.rootPath;
-                        break;
-                    default:
-                        exit(0);
-                }
+                await this.setWorkspace();
             }
 
-            await this.getGitDetailsFromRepository(this.workspacePath);
+            await this.getGitDetailsFromRepository();
         }
         catch (error) {
             this.telemetryHelper.logError(Layer, TracePoints.GetSourceRepositoryDetailsFailed, error);
             throw error;
         }
     }
 
-    private async getGitDetailsFromRepository(workspacePath: string): Promise<void> {
-        this.localGitRepoHelper = await LocalGitRepoHelper.GetHelperInstance(workspacePath);
-        this.inputs.sourceRepository = await this.localGitRepoHelper.getGitRepoDetails(workspacePath);
+    private async setWorkspace(): Promise<void> {
+        let workspaceFolders = vscode.workspace && vscode.workspace.workspaceFolders;
+        if (workspaceFolders && workspaceFolders.length > 0) {
+            this.telemetryHelper.setTelemetry(TelemetryKeys.SourceRepoLocation, Messages.selectWorkspaceFolder);
```

Comment:
rather than Messages which is subjected to change user a fixed telemetry action point.

Reply:
Agree. setting SourceRepoLocation value as 'workspace' / 'browse'. Also, adding a new telemetry key to log if there are multiple workspace folders :)